src/Shared/PackageManagers/NuGetProjectManager.cs

```diff
@@ -208,77 +144,26 @@ public override async Task<bool> PackageExists(PackageURL purl, bool useCache =
-                List<string> versionList = new();
-                foreach (JsonElement catalogPage in doc.RootElement.GetProperty("items").EnumerateArray())
-                {
-                    if (catalogPage.TryGetProperty("items", out JsonElement itemElement))
-                    {
-                        foreach (JsonElement item in itemElement.EnumerateArray())
-                        {
-                            JsonElement catalogEntry = item.GetProperty("catalogEntry");
-                            string? version = catalogEntry.GetProperty("version").GetString();
-                            if (version != null)
-                            {
-                                Logger.Debug("Identified {0} version {1}.", packageName, version);
-                                versionList.Add(version);
-                            }
-                            else
-                            {
-                                Logger.Debug("Identified {0} version NULL. This might indicate a parsing error.", packageName);
-                            }
-                        }
-                    }
-                    else
-                    {
-                        string? subDocUrl = catalogPage.GetProperty("@id").GetString();
-                        if (subDocUrl != null)
-                        {
-                            JsonDocument subDoc = await GetJsonCache(httpClient, subDocUrl);
-                            foreach (JsonElement subCatalogPage in subDoc.RootElement.GetProperty("items").EnumerateArray())
-                            {
-                                JsonElement catalogEntry = subCatalogPage.GetProperty("catalogEntry");
-                                string? version = catalogEntry.GetProperty("version").GetString();
-                                Logger.Debug("Identified {0} version {1}.", packageName, version);
-                                if (version != null)
-                                {
-                                    Logger.Debug("Identified {0} version {1}.", packageName, version);
-                                    versionList.Add(version);
-                                }
-                                else
-                                {
-                                    Logger.Debug("Identified {0} version NULL. This might indicate a parsing error.", packageName);
-                                }
-                            }
-                        }
-                        else
-                        {
-                            Logger.Debug("Catalog identifier was null.");
-                        }
-                    }
-                }
-                return SortVersions(versionList.Distinct());
+                // Sort versions, highest first, lowest last.
+                return SortVersions(versions);
```

Comment:
It looks like this sorting behavior is not documented on the parent type. Should this ordering be documented on the parent type? Do other implementations of this method for the parent type also follow this same ordering

Reply:
They all follow this same ordering, with the exception (in another open PR) for NPM. Where the ordering still applies, but if the version tagged `latest` isn't the highest version according to the sort, it gets removed from it's position in the list and placed at the beginning of the list.